client/my-sites/email/email-management/email-home.tsx

```diff
@@ -96,6 +98,7 @@ const EmailHome = ( props: EmailManagementHomeProps ) => {
 		return canCurrentUser( state, selectedSite.ID, 'manage_options' );
 	} );
 	const hasSitesLoaded = useSelector( hasLoadedSites );
+	const isAllDomainManagementEnabled = isEnabled( 'calypso/all-domain-management' );
```

Comment:
I think this may create an unwanted side-effect in other pages where the email-home component is used once this flag is enabled.

Our goal is to hide the back button only when email-home is being shown from the all-domain management page. But this will cause the back button to be hidden from everywhere.

Here's an example -

Before activating the flag -
<img width="1646" alt="image" src="https://github.com/user-attachments/assets/4cfca8b7-94e5-411f-84bb-5937135ea7a7">

After activating the flag - 

<img width="1646" alt="image" src="https://github.com/user-attachments/assets/4ba2f249-f8b4-462c-b97b-dbb35145a2c8">

Reply:
It's true, but as I understood the request, once we enable the feature flag, we want to replace the old experience with the new one. I thought that the old experience would not be accessible anymore.

Is this hypothesis true?

Reply:
That's a very valid thought. But probably those screens will be retired later, not within the scope of this iteration. You can read more about the discussion here - pcYYhz-2fY-p2#comment-2330 . This will also be safer, in case there's a bug in any of the new flows we introduce. Once these screens are rock solid and become more familiar to the users, we can retire the current screens.

Reply:
Hey @Imran92,

The regression issue has been resolved by replacing the feature flag with the context of the `HomeEmail` component usage.
Please, re-review it again once you find a few minutes.

Reply:
@Imran92 it makes sense to avoid changing behavior on other screens. However, I tested the "Back" button and it didn't work for me either on staging or in this branch. Then I click it, it just renders the same page but without the back button.

Reply:
@wojtekn, interesting. I just tested that scenario, and the back button also works within this branch and staging.
So, it should not be a blocker merging this PR.